--- atm/Rules.make~	Wed Dec 20 23:34:23 2000
+++ atm/Rules.make	Wed Dec 20 23:39:47 2000
@@ -60,7 +60,7 @@
 CFLAGS_LEX=$(CFLAGS_NOWARN) $(CFLAGS_OPT)
 CFLAGS_YACC=$(CFLAGS_NOWARN) $(CFLAGS_OPT) -DYY_USE_CONST
 LDFLAGS=
-LDLIBS += -L$(TOPDIR)/lib -latm
+LDLIBS += -lresolv -L$(TOPDIR)/lib -latm
 LIBDEPS += $(TOPDIR)/lib/libatm.a
 YACC=bison -y -d #-v
 INSTROOT=
--- atm/maint/Makefile~	Fri Apr  7 16:21:25 2000
+++ atm/maint/Makefile	Wed Dec 20 23:43:24 2000
@@ -9,4 +9,4 @@
 
 saaldump:	saaldump.o
 		$(CC) $(LDFLAGS) -o saaldump saaldump.o ../qgen/qd.dump.o \
-		  -L../saal -lsaal -L../lib -latm -latmd
+		  -lresolv -L../saal -lsaal -L../lib -latm -latmd
--- atm/ilmid/Makefile~	Wed Dec 20 23:34:23 2000
+++ atm/ilmid/Makefile	Wed Dec 20 23:45:44 2000
@@ -1,4 +1,4 @@
-LIBS = -L../lib -latmd -latm -Lasn1 -lasn1
+LIBS = -lresolv -L../lib -latmd -latm -Lasn1 -lasn1
 LIBDEPS=../lib/libatmd.a ../lib/libatm.a
 INCLUDES = -I../lib -Iasn1
 SRCS = rfc1157_snmp.c rfc1155_smi.c util.c io.c message.c \
--- atm/lib/Makefile~	Fri Jan 21 06:27:10 2000
+++ atm/lib/Makefile	Thu Dec 21 00:58:42 2000
@@ -1,10 +1,6 @@
 ATM_OBJS=text2atm.o atm2text.o atmequal.o sdu2cell.o text2qos.o qos2text.o \
   qosequal.o sap2text.o text2sap.o sapequal.o misc.o
-ifeq (/usr/lib/libresolv.a,$(wildcard /usr/lib/libresolv.a))
-ATM_OBJS += ans_l.o
-else
 ATM_OBJS += ans.o
-endif
 ATMD_OBJS=common.o diag.o kptr.o text2ip.o timer.o unix.o
 PGMS=#test
 GENLIBS=libatm.a libatmd.a
